### ISSUE TITLE
Unify out results of LGBM_BoosterDumpModel and LGBM_BoosterSaveModel

### ISSUE THREAD

**Issue by StrikerRUS** [microsoft/LightGBM#2604]
At present these two functions produce different output content. For example, text representation contains parameters and feature importance, while JSON format - not. These functions should produce the same content but in different formats. It will help many third-party libraries that work with LightGBM model's dumps.

```
import json

import numpy as np
import lightgbm as lgb

X = np.random.random((100, 2))
y = np.random.random(100)
lgb_data = lgb.Dataset(X, y)
bst = lgb.train({}, lgb_data, num_boost_round=3)
bst.save_model('save_model.txt')
with open('dump_model.json', 'w') as json_dump:
    json.dump(bst.dump_model(), json_dump, indent=2)
```

![image](https://user-images.githubusercontent.com/25141164/69920740-8f4f1600-149c-11ea-8c48-3ef8828d7475.png)

![image](https://user-images.githubusercontent.com/25141164/69920742-9413ca00-149c-11ea-8074-d93abbf8cf1e.png)

Unify:
- [x] feature importance (#2656)
- [x] feature infos (#2660)
- [ ] parameters
- [ ] trees' structure

**Comment by StrikerRUS** [comment]
Closed in favor of being in #2302. We decided to keep all feature requests in one place.

Welcome to contribute this feature! Please re-open this issue (or post a comment if you are not a topic starter) if you are actively working on implementing this feature.

**Comment by jameslamb** [comment]
This was locked accidentally. I just unlocked it. We'd still welcome contributions related to this feature!
